pkg/infrastructure/aws/clusterapi/iam.go

```diff
@@ -0,0 +1,189 @@
+package clusterapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/pkg/errors"
```

Comment:
Let's not add this dep :). You should use the stdlib `errors` for the `errors.As` calls.